src/main/java/build/buildfarm/cas/cfc/CASFileCache.java

```diff
@@ -1213,6 +1218,26 @@ public StartupCacheResults start(
     Duration startupTime = Duration.between(startTime, endTime);
     logger.log(Level.INFO, "Startup Time: " + startupTime.getSeconds() + "s");
 
+    // Start metrics collection thread
+    prometheusMetricsThread =
+        new Thread(
+            () -> {
+              while (!Thread.currentThread().isInterrupted()) {
+                try {
+                  casSizeMetric.set(size());
+                  casEntryCountMetric.set(entryCount());
+                  TimeUnit.SECONDS.sleep(30);
```

Comment:
I think since the change here will be fairly gradual we probably don't need to have this push every 30 seconds. I would say 1 or maybe even 5 minutes would be more than sufficient.